aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/MultilabelModel.scala

```diff
@@ -0,0 +1,387 @@
+  *
+  * @param modelId An identifier for the model.  Used in score and error reporting.
+  * @param featureNames feature names (parallel to featureFunctions)
+  * @param featureFunctions feature extracting functions.
+  * @param labelsInTrainingSet a sequence of all labels encountered during training. Note: the
+  *                            order of labels may relate to the predictor produced by
+  *                            predictorProducer.  It is the caller's responsibility to ensure
+  *                            the order is correct.  To mitigate such problems, both labels
+  *                            and indices into labelsInTrainingSet are passed to the predictor
+  *                            produced by predictorProducer.
+  * @param labelsOfInterest if provided, a sequence of labels will be extracted from the example
+  *                         for which a prediction is desired. The ''intersection'' of the
+  *                         extracted labels and the training labels will be the labels for which
+  *                         predictions will be produced.
+  * @param predictorProducer the function produced when calling this function is responsible for
+  *                          getting the data into the correct type and using it within an
+  *                          underlying ML library to produce a prediction.  The mapping back to
+  *                          (K, Double) pairs is also its responsibility.  If the predictor
+  *                          produced by predictorProducer is Closeable, it will be closed when
+  *                          MultilabelModel's close method is called.
+  * @param numMissingThreshold if provided, we check whether the threshold is exceeded.  If so,
+  *                            return an error instead of the computed score.  This is for missing
+  *                            data situations.
+  * @param auditor transforms a `Map[K, Double]` to a `B`.  Reports successes and errors.
+  * @tparam U upper bound on model output type `B`
+  * @tparam K type of label or class
+  * @tparam A input type of the model
+  * @tparam B output type of the model.
+  */
+case class MultilabelModel[U, K, -A, +B <: U](
+    modelId: ModelIdentity,
+    featureNames: sci.IndexedSeq[String],
+    featureFunctions: sci.IndexedSeq[GenAggFunc[A, Sparse]],
+    labelsInTrainingSet: sci.IndexedSeq[K],
+    labelsOfInterest: Option[GenAggFunc[A, sci.IndexedSeq[K]]],
+    predictorProducer: SparsePredictorProducer[K],
+    numMissingThreshold: Option[Int],
+    auditor: Auditor[U, Map[K, Double], B])
+extends SubmodelBase[U, Map[K, Double], A, B]
+   with RegressionFeatures[A] {
+
+  import MultilabelModel._
+
+  /**
+    * predictory is transient lazy value because we don't need to worry about serialization.
+    * We don't care about the lazy property.  It should be created eagerly.
+    */
+  @transient private[this] lazy val predictor = predictorProducer()
+  predictor // Force predictor eagerly
```

Comment:
Why bother with `lazy` here?  Wouldn't this be the same as the above two lines

`@transient private[this] val predictor = predictorProducer()`

Reply:
Has to be lazy.

```scala
import java.io.{
  ByteArrayInputStream, 
  ByteArrayOutputStream, 
  ObjectInputStream,
  ObjectOutputStream}

case class Fail(i: Int) {
  @transient val s = i.toString
}

case class Success(i: Int) {
  @transient lazy val s = i.toString
}

def serializeRoundTrip[A <: AnyRef](a: A) = {
  val baos = new ByteArrayOutputStream
  val oos = new ObjectOutputStream(baos)
  oos.writeObject(a)
  oos.close
  val ois = new ObjectInputStream(new ByteArrayInputStream(baos.toByteArray))
  ois.readObject().asInstanceOf[A]
}

val fail = serializeRoundTrip(Fail(1))
assert(fail.s == null)

val success = serializeRoundTrip(Success(1))
assert(success.s == "1")
```